Sources_v3/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift

```diff
@@ -0,0 +1,222 @@
+    
+    open func setupAppearance() {
+        titleLabel.numberOfLines = 1
+        titleLabel.font = .boldSystemFont(ofSize: .scaledFontSize(for: 14))
+        
+        subtitleLabel.numberOfLines = 1
+        subtitleLabel.textColor = .gray
+        subtitleLabel.font = .systemFont(ofSize: .scaledFontSize(for: 12.5))
+        
+        timestampLabel.numberOfLines = 1
+        timestampLabel.textColor = .gray
+        timestampLabel.font = .systemFont(ofSize: .scaledFontSize(for: 12),
+                                          weight: .medium)
+    }
+    
+    open func setupLayout() {
+        embed(container)
+        
+        container.leftStackView.isHidden = false
+        container.leftStackView.alignment = .center
+        
+        container.leftStackView.addArrangedSubview(avatarView)
+        
+        container.centerStackView.isHidden = false
+        container.centerStackView.axis = .vertical
+        container.centerStackView.spacing = 2
+        
+        container.centerStackView.addArrangedSubview(titleLabel)
+        container.centerStackView.addArrangedSubview(subtitleLabel)
+        
+        container.rightStackView.isHidden = false
+        container.rightStackView.axis = .vertical
+        container.rightStackView.spacing = 6
+        
+        let rightTopContainer = UIStackView()
+        
+        rightTopContainer.addSpacer()
+        rightTopContainer.addArrangedSubview(unreadCountBubble)
+        
+        let rightBottomContainer = UIStackView()
+        rightBottomContainer.spacing = 6
+        
+        rightBottomContainer.addSpacer()
+        rightBottomContainer.addArrangedSubview(readStatusView)
+        rightBottomContainer.addArrangedSubview(timestampLabel)
+        
+        container.rightStackView.addSpacer()
+        container.rightStackView.addArrangedSubview(rightTopContainer)
+        container.rightStackView.addArrangedSubview(rightBottomContainer)
+    
+        avatarView.widthAnchor.constraint(equalToConstant: 56).isActive = true
+        readStatusView.widthAnchor.constraint(equalToConstant: 16).isActive = true
+        unreadCountBubble.heightAnchor.constraint(equalToConstant: 16).isActive = true
+        
+        container.layoutIfNeeded()
+        
+        widthAnchor.constraint(equalToConstant: UIScreen.main.bounds.width).isActive = true
+    }
+    
+    open override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        container.layoutMargins = .init(top: 0, left: 0, bottom: 0, right: 9)
+        container.isLayoutMarginsRelativeArrangement = true
+        
+        container.centerStackView.addSideMarginsToMatch(height: frame.height)
+        
+        container.rightStackView.layoutMargins = .init(top: 0, left: 0, bottom: container.centerStackView.layoutMargins.bottom, right: 0)
+        container.rightStackView.isLayoutMarginsRelativeArrangement = true
+    }
+    
+    open func updateContent() {
+        guard let channel = channel else { return }
```

Comment:
I'd say If `channel` doesn't exist the content shouldn't be preserved but reset.